clients/src/main/java/org/apache/kafka/common/utils/Checksums.java

```
@@ -30,6 +33,18 @@
  */
 public final class Checksums {
 
+    private static final MethodHandle BYTE_BUFFER_UPDATE;
+
+    static {
+        MethodHandle byteBufferUpdate = null;
+        try {
+            byteBufferUpdate = MethodHandles.publicLookup().findVirtual(Checksum.class, "update", MethodType.methodType(void.class, ByteBuffer.class));
+        } catch (Throwable silent) {
```

I understand that you are keeping it here for backward compatibility with older versions (8) of JDK which don't have this API. But we should ideally want to throw an error when we remove support for JDK 8 from Kafka.

Could we do one of the following two options here:

1. (preferred) Use [isJava9Compatible](https://docs.gradle.org/current/javadoc/org/gradle/api/JavaVersion.html#isJava9Compatible--) to decide whether we want to propagate the exception here or fail silently OR
2. add a TODO here so that we can remember to propagate exception when support for JDK 8 is removed? 

Also, a comment explaining why are we consuming the exception would be nice.

---

I've added a LOG::warn in case the given signature is expected to be found (although it doesn't consider the security manager enabling option - but I don't believe is a common Kafka practice suggested to users).